include/nuttx/atexit.h

```diff
@@ -0,0 +1,86 @@
+/****************************************************************************
+ * include/nuttx/atexit.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __INCLUDE_NUTTX_ATEXIT_H
+#define __INCLUDE_NUTTX_ATEXIT_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <stdlib.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/* Amount of exit functions */
+
+#define ATEXIT_MAX (CONFIG_LIBC_MAX_EXITFUNS)
+
+/****************************************************************************
+ * Public Types
+ ****************************************************************************/
+
+enum atexit_type_e
+{
+  ATTYPE_NONE,
+  ATTYPE_ATEXIT,
+  ATTYPE_ONEXIT,
+  ATTYPE_CXA
+};
+
+struct atexit_s
+{
+  int         type;
+  CODE void (*func)(void);
+  FAR void   *arg;
+};
+
+struct atexit_list_s
+{
+  int             next;
```

Comment:
maybe we can rename to `nfuncs` or `count`. The `next` is not obvious.

Reply:
The original name was idx, I can name this to whatever.

Reply:
Let's rename to `nfuncs`. I skipped the header and initially during the code review was assuming that `next` is a pointer, so had to go back to header and check.

Reply:
Done